pyscript.core/src/python/pystdlib/display.py

```diff
@@ -1,10 +1,14 @@
 # ⚠️ WARNING - both `document` and `window` are added at runtime
 
+# XXX antocuni: I think this is wrong: it works in the main thread but not in
+# the worker, because the rest of the code expects window and document to be
+# proxies (see workerPyScriptModule in core.js)
 import base64
 import html
 import io
 import re
 
+from js import document, window
```

Comment:
this breaks in workers ... this file had a warning because it's an artifact with different `window` and `document` accordingly with the environment ... there was a test to demonstrate the current logic already worked.

Reply:
Right. This shouldn't be the final import (it's here as as placeholder). The idea is that it'd be something like 

`from pyscript import document, window` and we leave the `pyscript` module deal with the dichotomy of `main` thread vs `worker`. One idea is that if there's a flag or something, the import on the `pyscript` module side could be something like

```python
if __in_worker__:
    # this or something like from polyscript.xworker import window, sync, ....
    document=polyscript.xworker.window.document
    window=polyscript.xworker.window
    sync=polyscript.xworker.sync
else:
    from js import window, document
    sync = None
```

Reply:
that's already the case ... the artifact is executed as string and those variables are already defined within the module as string. There is no need to do that in Python when Python code is executed already either in Worker or Main and the logic perfectly knows when that's the case. No global leaks for `__in_worker__` needed too.

Reply:
yes, the main point of the PR is to avoid executing the artifact as a string, because it creates lots of problems on the python side.
Ideally, we would like to have a `pyscript` package which contains a bunch of Python files and `import` them, instead of using `runPython`.